FreeRTOS_IP.c

```diff
@@ -1853,10 +1843,11 @@ static eFrameProcessingResult_t prvAllowIPPacket( const IPPacket_t * const pxIPP
              * This method may decrease the usage of sparse network buffers. */
             uint32_t ulDestinationIPAddress = pxIPHeader->ulDestinationIPAddress;
 
-            /* Ensure that the incoming packet is not fragmented (only outgoing
-             * packets can be fragmented) as these are the only handled IP frames
-             * currently. */
-            if( ( pxIPHeader->usFragmentOffset & ipFRAGMENT_OFFSET_BIT_MASK ) != 0U )
+            /* Ensure that the incoming packet is not fragmented. This stack
+             * doesn't not support IP fragmentation and always sets the "don't fragment"
+             * flag on outgoing IP frames. The first fragment coming in will have its
+             * "more fragments" flag set and later fragments will have a non-zero offset. */
```

Comment:
I like the comment here. Very helpful and explanatory. But I think that we might be able to make it a bit more accurate as so:

```suggestion
             * flag on outgoing IP frames. All but the last fragment coming in will have their
             * "more fragments" flag set and the last fragment will have a non-zero offset. */
```

What do you think?